### DISCUSSION TITLE
How to make apps and ports for onions os?

### DISCUSSION THREAD

**Discussion by BookCatKid** [OnionUI/Onion#1382] [Q&A] [answered]
How do I make a port (pygame or other) for onions os? Is there a guide somewhere? Also apps too, but those are probably a lot more complicated. I have a pretty good knowledge of linux, python, c/c++...
There should be a guide in the wiki on how to do this.

**Comment by Aemiii91** [comment, marked as answer]
There's a guide on how to setup ports here: <https://onionui.github.io/docs/ports#how-to-add-your-own-ports-for-advanced-users>

**Reply by BookCatKid** [reply]
Is there a very basic simple example of a pygame port?

**Reply by Aemiii91** [reply]
The Snake one is pretty basic: <https://github.com/OnionUI/Ports-Collection/tree/main/Snake%20(PyGame)/Roms/PORTS/Games/Snake%20(PyGame)>

**Reply by BookCatKid** [reply]
Is it possible to add additional libraries to the built in python installation?

**Reply by Aemiii91** [reply]
I don't see why not.

**Reply by BookCatKid** [reply]
So, How? What command would I run?

**Reply by Aemiii91** [reply]
I have no idea personally, but I would imagine you'd just add the libs here: [`.tmp_update/lib/parasyte/python2.7`](https://github.com/OnionUI/Onion/tree/main/static/build/.tmp_update/lib/parasyte/python2.7)
